src/components/Admin/UserPage/DeleteOldDataModal.tsx

```diff
@@ -0,0 +1,147 @@
+import {
+  Box,
+  Button,
+  Checkbox,
+  CheckboxGroup,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+  Stack,
+  useDisclosure,
+  Text,
+  Divider,
+  Flex,
+} from '@chakra-ui/react';
+import React, {useState} from 'react';
+import {useTranslation} from 'react-i18next';
+import {FaTrash} from 'react-icons/fa';
+import {useAuth} from '../../../contexts/AuthContext';
+import {firestore} from '../../../firebase';
+
+function DeleteOldDataModal(): JSX.Element {
+  // --------------- State maintenance variables ------------------------
+  const {isOpen, onOpen, onClose} = useDisclosure();
+  const [areDisclosuresChecked, setAreDisclosuresChecked] = useState([
+    false,
+    false,
+    false,
+  ]);
+
+  const allDisclosuresChecked = areDisclosuresChecked.every(Boolean);
+  const {isAdmin} = useAuth();
+
+  const {t} = useTranslation('administration');
+
+  /**
+   * Resets modal state values before closing the modal.
+   */
+  function handleClose(): void {
+    // Modal state
+    setAreDisclosuresChecked([false, false, false]);
+    onClose();
+  }
+
+  /**
+   * Marks data older than a defined time as ready for deletion by a Cloud Function
+   */
+  async function markOldDataForDeletion(): Promise<void> {
+    if (allDisclosuresChecked && isAdmin) {
+      const sensorsList = await firestore.collection('sensors').get();
+      const deletionDocRef = firestore.collection('deletion').doc('todo');
+
+      // Get current mapping or set to empty
+      const mapping =
+        (await (await deletionDocRef.get())?.data()?.data) ??
```

Comment:
To do this, we'll need to give admins read access

Reply:
I think we do need that though, in order to avoid overwriting things

Reply:
We did that already in #547